lib/src/models/binned.dart

```diff
@@ -709,61 +842,63 @@ class BinnedChartPainter extends CustomPainter {
   @override
   void paint(Canvas canvas, Size size) {
     // Calculate the projection used for all points in the series
-    Size plotSize = Size(size.width - tickLabelMargin.left - tickLabelMargin.right,
+    final Size plotSize = Size(size.width - tickLabelMargin.left - tickLabelMargin.right,
         size.height - tickLabelMargin.top - tickLabelMargin.bottom);
-    Rect plotWindow = Offset(tickLabelMargin.left, tickLabelMargin.top) & plotSize;
-    Offset offset = Offset(tickLabelMargin.left, tickLabelMargin.top);
+    final Rect plotWindow = Offset(tickLabelMargin.left, tickLabelMargin.top) & plotSize;
+    final Offset offset = Offset(tickLabelMargin.left, tickLabelMargin.top);
 
     // Since all of the objects in the series use the same marker style,
     // we can calculate the [Paint] objects once and reuse them.
-    for (BinnedDataContainer bins in binContainers.values) {
-      for (int i = 0; i < bins.bins.length; i++) {
-        BinnedData bin = bins.bins[i];
-        Rect binRect = bin.rectToPixel(
-          axes: axes,
-          chartSize: plotSize,
-          offset: offset,
-          mainAxisAlignment: mainAxisAlignment,
-        );
+    // Loop through each series in the binContainers
+    binContainers.forEach((seriesIndex, container) {
```

Comment:
I think that the dart idiomatic preference is to use `for` loops, not `forEach` statements anymore.